src/EventListener/ReadListener.php

```diff
@@ -28,11 +30,17 @@
 {
     private $collectionDataProvider;
     private $itemDataProvider;
+    private $subresourceDataProvider;
 
-    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider)
+    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null)
     {
         $this->collectionDataProvider = $collectionDataProvider;
         $this->itemDataProvider = $itemDataProvider;
+        $this->subresourceDataProvider = $subresourceDataProvider;
+
+        if (null === $subresourceDataProvider) {
```

Comment:
You can keep this exception. It is only triggered when a subresource is encountered, sounds legit to me.